lib/pengine/rules.c

```diff
@@ -1032,7 +1032,11 @@ pe__eval_attr_expr(xmlNodePtr expr, pe_rule_eval_data_t *rule_data)
     if (value != NULL && h_val != NULL) {
```

Comment:
Next push will break the comparison logic into two helper functions (one called by the other).

- Gives us an easier way to default to string comparison if numeric parse fails (call the function again with the type argument set to "string").
- Easier to read.